homeassistant/components/vconnex/entity.py

```diff
@@ -0,0 +1,202 @@
+        """Set additional param value dict."""
+        self.set_additional_param_value(param_values)
+        return self
+
+    def with_additional_param_func(
+        self, func: Callable[[dict, VconnexDevice], dict]
+    ) -> EntityDescResolver:
+        """Set additional param function."""
+        self._additional_param_func = func
+        return self
+
+    def from_param(self, param_dict: dict, device: VconnexDevice = None) -> T | None:
+        """Get Entity Description from param dict."""
+        new_param_dict = self._process_param(param_dict, device)
+        if new_param_dict is not None:
+            return self.__cls(**new_param_dict)
+        return None
+
+    @staticmethod
+    def of(cls_type: type[T]) -> EntityDescResolver:  # pylint: disable=invalid-name
+        """Create Entity Description Resolver instance."""
+        return EntityDescResolver(cls_type)
+
+
+class EntityDescListResolver:
+    """Entity Description List Resolver."""
+
+    def __init__(
+        self,
+        device_types: set[int],
+        param_types: set[int],
+        resolver: EntityDescResolver,
+    ) -> None:
+        """Create Entity Description List Resolver object."""
+        self._accept_device_types = device_types
+        self._accept_param_types = param_types
+        self._entity_desc_resolver: EntityDescResolver = resolver
+
+    def from_device(self, device: VconnexDevice) -> list:
+        """Get Description Entity List from device."""
+        if int(device.deviceTypeCode) in self._accept_device_types:
+            if device is not None and len(param_list := device.params) > 0:
+                description_list = []
+                for param in param_list:
+                    if int(param.get("type", 0)) in self._accept_param_types:
+                        description = self._entity_desc_resolver.from_param(
+                            param, device
+                        )
+                        if description is not None:
+                            description_list.append(description)
+                return description_list
+        return []
+
+
+class VconnexEntity(Entity):
+    """Vconnex Entity."""
+
+    def __init__(
+        self,
+        device: VconnexDevice,
+        device_manager: VconnexDeviceManager,
+        description: EntityDescription,
+    ) -> None:
+        """Create base entity object."""
+        self.device = device
+        self.device_manager = device_manager
+        self.entity_description = description
+
+        self._attr_unique_id = f"{DOMAIN}.{device.deviceId}"
+
+    @property
+    def name(self) -> str:
+        """Get entity name."""
+        return (
+            f"[{self.device.name}] {self.entity_description.name}"
+            if (
+                self.entity_description is not None
+                and self.entity_description.name is not None
+            )
+            else self.device.name
+        )
+
+    @property
+    def device_info(self) -> DeviceInfo:
```

Comment:
Is the information in `self.device` available during initialisation? If yes, then it could be set in `self._attr_device_info` inside `__init__()` instead of using a property.